### ISSUE TITLE
SC2139 arises for no reason

### ISSUE THREAD

**Issue by zzJinux** [koalaman/shellcheck#3268]
#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): SC2139
- My shellcheck version (`shellcheck --version` or "online"): 0.10.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
alias {eee,fff}='echo hi'
```

#### Here's what shellcheck currently says:
```
Line 2:
alias {eee,fff}='echo hi'
      ^-- SC2139 (warning): This expands when defined, not when used. Consider escaping.
```

#### Here's what I wanted or expected to see:
```
No issues detected!
```

**Comment by wileyhy** [comment]
Hello,

It's a little unclear from your post just what exactly you wanted for the
example code to have accomplished.

As-is, the `alias` command would create two separate and functionally
equivalent aliases, `eee` and `fff`. However by enclosing the entire
definition string in single quotes, it's also possible to create the alias
`{eee,fff}`, although this is rarely intended.

In either case, aliases are usually disabled by default in shell scripts,
so it would be necessary to enable them via `shopt -s expand_aliases`.

~ $ set -x; shopt | grep alias; alias {eee,fff}='echo hi'
+ grep alias
+ shopt
expand_aliases on
progcomp_alias off
+ alias 'eee=echo hi' 'fff=echo hi'
~ $ alias '{eee,fff}=echo hi' + alias '{eee,fff}=echo hi'
~ $ alias
+ alias
alias eee='echo hi'
alias fff='echo hi'
alias {eee,fff}='echo hi'
~ $ {eee,fff}
+ echo hi
hi
~ $

Some would discourage use of aliases in shell scripts, as for nearly all
uses aliases have been superceded by functions.

While both of these cases, re `alias`, would be valid code, it's much more
common for people to want for any expansions to occur when an alias is
called, rather than when an alias is defined.

In terms of brace expansion on the left side of any assignment statement,
there are numerous cases where this would produce an error.

~ $ declare "{foo,bar}=quux"
+ declare '{foo,bar}=quux'
bash: declare: `{foo,bar}=quux': not a valid identifier
~ $ {a,b}=( c d e )
bash: syntax error near unexpected token `c'

Any shellcheck tests can be disabled by including a "disable statement" in
your script:

# shellcheck disable=SC2139

**Comment by zzJinux** [comment]
I manage dotfiles in a git repo. The example is a simplified version of a script in the dotfiles repo. `alias foo='echo hi'` passes, so why `alias {eee,fff}='echo hi'` shouldn't? At least the code `SC2139` is confusing.
